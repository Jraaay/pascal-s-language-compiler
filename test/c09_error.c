
int x, y; 

int main(int argc, char* argv[]){
	x = a > b; 
}
