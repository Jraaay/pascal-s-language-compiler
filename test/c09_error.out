{
    "ast": {
        "length": 5,
        "type": "programstruct",
        "program_head": {
            "length": 6,
            "type": "program_head",
            "ID": "example",
            "idlist": {
                "length": 4,
                "type": "idlist",
                "ids": [
                    "input",
                    "output"
                ]
            }
        },
        "program_body": {
            "length": 5,
            "type": "program_body",
            "const_declarations": null,
            "var_declarations": {
                "length": 4,
                "type": "var_declarations",
                "var_declaration": {
                    "length": 4,
                    "id": 0,
                    "type": "var_declaration",
                    "values": [
                        {
                            "idlist": {
                                "length": 4,
                                "type": "idlist",
                                "ids": [
                                    "x",
                                    "y"
                                ]
                            },
                            "type": {
                                "length": 2,
                                "type": "type",
                                "_type": {
                                    "type": "basic_type",
                                    "_type": "INTEGER"
                                }
                            }
                        }
                    ]
                }
            },
            "subprogram_declarations": {
                "length": 1,
                "type": "subprogram_declarations",
                "subprograms": []
            },
            "compound_statement": {
                "length": 4,
                "type": "compound_statement",
                "statement_list": {
                    "length": 4,
                    "type": "statement_list",
                    "statements": [
                        {
                            "length": 4,
                            "type": "statement",
                            "_type": "variable",
                            "variable": {
                                "length": 3,
                                "type": "variable",
                                "__type": "INTEGER",
                                "ID": "x",
                                "id_varpart": null
                            },
                            "ASSIGNOP": ":=",
                            "expression": {
                                "length": 4,
                                "type": "expression",
                                "__type": "BOOLEAN",
                                "simple_expression_1": {
                                    "length": 2,
                                    "type": "simple_expression",
                                    "__type": "UNDEFINED",
                                    "term": {
                                        "length": 2,
                                        "type": "term",
                                        "__type": "UNDEFINED",
                                        "factor": {
                                            "length": 2,
                                            "type": "factor",
                                            "_type": "variable",
                                            "__type": "UNDEFINED",
                                            "variable": {
                                                "length": 3,
                                                "type": "variable",
                                                "__type": "UNDEFINED",
                                                "ID": "a",
                                                "id_varpart": null
                                            }
                                        }
                                    }
                                },
                                "RELOP": ">",
                                "simple_expression_2": {
                                    "length": 2,
                                    "type": "simple_expression",
                                    "__type": "UNDEFINED",
                                    "term": {
                                        "length": 2,
                                        "type": "term",
                                        "__type": "UNDEFINED",
                                        "factor": {
                                            "length": 2,
                                            "type": "factor",
                                            "_type": "variable",
                                            "__type": "UNDEFINED",
                                            "variable": {
                                                "length": 3,
                                                "type": "variable",
                                                "__type": "UNDEFINED",
                                                "ID": "b",
                                                "id_varpart": null
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    ]
                }
            }
        }
    },
    "symbolTable": {
        "constants": [],
        "variables": [
            {
                "id": 0,
                "token": "x",
                "type": "INTEGER",
                "isArray": false,
                "dimension": 0,
                "size": [],
                "start": [],
                "recordTable": null
            },
            {
                "id": 1,
                "token": "y",
                "type": "INTEGER",
                "isArray": false,
                "dimension": 0,
                "size": [],
                "start": [],
                "recordTable": null
            }
        ],
        "subFunc": []
    },
    "error": [
        {
            "code": "C-02",
            "info": {
                "line": 5,
                "value": "a",
                "lexpos": 70
            }
        },
        {
            "code": "C-02",
            "info": {
                "line": 5,
                "value": "b",
                "lexpos": 73
            }
        },
        {
            "code": "C-09",
            "info": {
                "line": 5,
                "lexpos": 73
            }
        }
    ],
    "warning": []
}